Grbl_Esp32/src/System.cpp

```diff
@@ -46,42 +47,46 @@ bool         debouncing = false;  // debouncing in process
 void system_ini() {  // Renamed from system_init() due to conflict with esp32 files
     // setup control inputs
 
-#ifdef CONTROL_SAFETY_DOOR_PIN
-    pinMode(CONTROL_SAFETY_DOOR_PIN, INPUT_PULLUP);
-    attachInterrupt(digitalPinToInterrupt(CONTROL_SAFETY_DOOR_PIN), isr_control_inputs, CHANGE);
-#endif
-#ifdef CONTROL_RESET_PIN
-    pinMode(CONTROL_RESET_PIN, INPUT_PULLUP);
-    attachInterrupt(digitalPinToInterrupt(CONTROL_RESET_PIN), isr_control_inputs, CHANGE);
-#endif
-#ifdef CONTROL_FEED_HOLD_PIN
-    pinMode(CONTROL_FEED_HOLD_PIN, INPUT_PULLUP);
-    attachInterrupt(digitalPinToInterrupt(CONTROL_FEED_HOLD_PIN), isr_control_inputs, CHANGE);
-#endif
-#ifdef CONTROL_CYCLE_START_PIN
-    pinMode(CONTROL_CYCLE_START_PIN, INPUT_PULLUP);
-    attachInterrupt(digitalPinToInterrupt(CONTROL_CYCLE_START_PIN), isr_control_inputs, CHANGE);
-#endif
-#ifdef MACRO_BUTTON_0_PIN
-    grbl_msg_sendf(CLIENT_SERIAL, MsgLevel::Info, "Macro Pin 0");
-    pinMode(MACRO_BUTTON_0_PIN, INPUT_PULLUP);
-    attachInterrupt(digitalPinToInterrupt(MACRO_BUTTON_0_PIN), isr_control_inputs, CHANGE);
-#endif
-#ifdef MACRO_BUTTON_1_PIN
-    grbl_msg_sendf(CLIENT_SERIAL, MsgLevel::Info, "Macro Pin 1");
-    pinMode(MACRO_BUTTON_1_PIN, INPUT_PULLUP);
-    attachInterrupt(digitalPinToInterrupt(MACRO_BUTTON_1_PIN), isr_control_inputs, CHANGE);
-#endif
-#ifdef MACRO_BUTTON_2_PIN
-    grbl_msg_sendf(CLIENT_SERIAL, MsgLevel::Info, "Macro Pin 2");
-    pinMode(MACRO_BUTTON_2_PIN, INPUT_PULLUP);
-    attachInterrupt(digitalPinToInterrupt(MACRO_BUTTON_2_PIN), isr_control_inputs, CHANGE);
-#endif
-#ifdef MACRO_BUTTON_3_PIN
-    grbl_msg_sendf(CLIENT_SERIAL, MsgLevel::Info, "Macro Pin 3");
-    pinMode(MACRO_BUTTON_3_PIN, INPUT_PULLUP);
-    attachInterrupt(digitalPinToInterrupt(MACRO_BUTTON_3_PIN), isr_control_inputs, CHANGE);
-#endif
+    if (ControlSafetyDoorPin->get() != Pin::UNDEFINED) {
+        ControlSafetyDoorPin->get().setAttr(Pin::Attr::Input | Pin::Attr::PullUp);
+        ControlSafetyDoorPin->get().attachInterrupt(isr_control_inputs, CHANGE);
+    }
+
+    if (ControlResetPin->get() != Pin::UNDEFINED) {
```

Comment:
Had to make the  follow temporary changes to system_ini() to stop the crashes until I deleted the pin definition from the command line.
if (ControlResetPin->get() != Pin::UNDEFINED) {
        ControlResetPin->get().setAttr(Pin::Attr::Input);// | Pin::Attr::PullUp);
//        ControlResetPin->get().attachInterrupt(isr_control_inputs, CHANGE);
    }
There is a definite issue with attachInterupt and an input-only pin.

Reply:
Ough. The main issue here is that PullUp is currently "enable the internal pullup resistor", instead of the "existance of a pullup resistor", which is different from how it was originally designed. Code should probably never specify pullup.